emp-wrapper/src/main/java/org/ballerinalang/sf/ListenerUtil.java

```diff
@@ -132,7 +170,67 @@ private static long getReplayFrom(ObjectValue service) {
+    public static BMap<BString, Object> toBMap(Map map) {
+        BMap<BString, Object> returnMap = ValueCreator.createMapValue();
+        if (map != null) {
+            for (Object aKey : map.keySet().toArray()) {
+                returnMap.put(StringUtils.fromString(aKey.toString()),
+                        StringUtils.fromString(map.get(aKey).toString()));
+            }
+        }
+        return returnMap;
+    }
+
+//    /**
+//     * Convert Map to Ballerina record tpe
+//     *
+//     * @param event Input Map used to convert to BMap.
+//     * @return Converted BMap object.
+//     */
+//    private static BMap<BString, Object> getEventDataRecord(Map<String, Object> event)  {
+//        Gson gson = new Gson();
+//        Object[] eventData = new Object[2];
+//        BMap<BString, Object> eventDataRecord =
+//                ValueCreator.createRecordValue(PACKAGE_ID_SFDC, EVENT_DATA_RECORD);
+//        eventData[0] = toBMap(event);
+//        BMap<BString, Object> eventMetadataRecord =
+//                ValueCreator.createRecordValue(PACKAGE_ID_SFDC, EVENT_METADATA_RECORD);
+//        Object[] metadata = new Object[8];
+//        Integer commitTimestamp = (Integer) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("commitTimestamp");
+//        metadata[0] = commitTimestamp;
+//        String transactionKey = (String) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("transactionKey");
+//        metadata[1] = transactionKey;
+//        String changeOrigin = (String) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("changeOrigin");
+//        metadata[2] = changeOrigin;
+//        String changeType = (String) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("changeType");
+//        metadata[3] = changeType;
+//        String entityName = (String) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("entityName");
+//        metadata[4] = entityName;
+//        Integer sequenceNumber = (Integer) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("sequenceNumber");
+//        metadata[5] = sequenceNumber;
+//        String commitUser = (String) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("commitUser");
+//        metadata[6] = commitUser;
+//        Integer commitNumber = (Integer) new JSONObject(gson.toJson(event.get("payload")))
+//                .getJSONObject("ChangeEventHeader").get("commitNumber");
+//        metadata[7] = commitNumber;
+//        eventData[1] = ValueCreator.createRecordValue(eventMetadataRecord, metadata);
+//        return ValueCreator.createRecordValue(eventDataRecord, eventData);
+//    }
```

Comment:
Shall we model this information to a record? Meta information can be modelled to a record.

Reply:
Provided the information returned from listener events as a record type.